components/family-studies/migrations/src/main/java/org/phenotips/studies/family/migrations/R71500PhenoTips2944DataMigration.java

```diff
@@ -0,0 +1,311 @@
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(71500);
+    }
+
+    @Override
+    public void hibernateMigrate() throws DataMigrationException, XWikiException
+    {
+        getStore().executeWrite(getXWikiContext(), this);
+    }
+
+    @Override
+    public Object doInHibernate(Session session) throws HibernateException, XWikiException
+    {
+        XWikiContext context = getXWikiContext();
+        XWiki xwiki = context.getWiki();
+        this.hgnc = this.vocabularies.getVocabulary(HGNC);
+
+        // Select all families
+        Query q = session.createQuery("select distinct o.name from BaseObject o where o.className = '"
+            + this.serializer.serialize(PEDIGREE_CLASS_REFERENCE)
+            + "' and o.name <> 'PhenoTips.FamilyTemplate'" + "' and o.name <> 'PhenoTips.PatientTemplate'");
+
+        @SuppressWarnings("unchecked")
+        List<String> docs = q.list();
+
+        this.logger.debug("Found {} family documents", docs.size());
+
+        for (String docName : docs) {
+            XWikiDocument familyXDocument;
+            boolean updated = false;
+
+            try {
+                familyXDocument = xwiki.getDocument(this.resolver.resolve(docName), context);
+                if (familyXDocument == null) {
+                    continue;
+                }
+
+                BaseObject pedigreeXObject = familyXDocument.getXObject(PEDIGREE_CLASS_REFERENCE);
+                if (pedigreeXObject == null) {
+                    continue;
+                }
+
+                this.logger.debug("Updating pedigree for family {}.", docName);
+                updated = this.updatePedigreeGenes(pedigreeXObject, context, docName);
+                if (!updated) {
+                    continue;
+                }
+
+                familyXDocument.setComment(this.getDescription());
+                familyXDocument.setMinorEdit(true);
+
+            } catch (Exception e) {
+                this.logger.error("Error converting gene data for family {}: [{}]", docName, e.getMessage());
+                continue;
+            }
+
+            try {
+                // There's a bug in XWiki which prevents saving an object in the same session that it was loaded,
+                // so we must clear the session cache first.
+                session.clear();
+                ((XWikiHibernateStore) getStore()).saveXWikiDoc(familyXDocument, context, false);
+                session.flush();
+            } catch (DataMigrationException e) {
+                //
+            }
+
+        }
+        return null;
+    }
+
+    private boolean updatePedigreeGenes(BaseObject pedigreeXObject, XWikiContext context, String documentName)
+    {
+        boolean updatedGenes = false;
+        boolean updatedColors = false;
+        String dataText = pedigreeXObject.getStringValue(PEDIGREECLASS_JSONDATA_KEY);
```

Comment:
I would put all of these inside a try-catch block, since pedigreeXObject.getStringValue may in theory throw something as well I think (or be null if higher-level code is incorrect). The boolean variablkes ar enot useful outside try-catch as well, as the last return can be inside